scripts/interopTestClient.ts

```diff
@@ -0,0 +1,244 @@
+                if (typeof bucketBoundary !== "string") {
+                    throw new Error("VDAF defeinition's `buckets` parameter is not an array of strings");
+                }
+            }
+            if (typeof body.measurement !== "string") {
+                throw new Error("Measurement is not a string");
+            }
+
+            vdafObject = {
+                type: vdaf.type,
+                buckets: vdaf.buckets as string[],
+            }
+            measurement = body.measurement;
+            break;
+
+        default:
+            throw new Error(`Unrecognized VDAF ${vdaf.type}`);
+    }
+
+    return {
+        taskId: body.taskId,
+        leader: body.leader,
+        helper: body.helper,
+        vdaf: vdafObject,
+        measurement: measurement,
+        nonceTime: body.nonceTime,
+        minBatchDuration: body.minBatchDuration,
+    }
+}
+
+function assertUnreachable(_: never): never {
+    throw new Error("Unreachable code");
+}
+
+async function uploadHandler(req: Request, res: Response): Promise<void> {
+    try {
+        const body = sanitizeRequest(req.body);
+
+        // TODO (issue #97): Implement a way to construct a report with a
+        // custom nonce timestamp.
+        if (body.nonceTime !== undefined) {
+            throw new Error("`nonceTime` is not yet supported");
+        }
+
+        switch (body.vdaf.type) {
+            case "Prio3Aes128Count":
+                await new DAPClient({
+                    taskId: body.taskId,
+                    leader: body.leader,
+                    helper: body.helper,
+                    minBatchDurationSeconds: body.minBatchDuration,
+                    type: "count",
+                }).sendMeasurement(body.measurement != 0);
+                break;
+
+            case "Prio3Aes128Sum":
+                await new DAPClient({
+                    taskId: body.taskId,
+                    leader: body.leader,
+                    helper: body.helper,
+                    minBatchDurationSeconds: body.minBatchDuration,
+                    type: "sum",
+                    bits: body.vdaf.bits,
+                }).sendMeasurement(Number(body.measurement));
+                break;
+
+            case "Prio3Aes128Histogram":
+                await new DAPClient({
+                    taskId: body.taskId,
+                    leader: body.leader,
+                    helper: body.helper,
+                    minBatchDurationSeconds: body.minBatchDuration,
+                    type: "histogram",
+                    buckets: body.vdaf.buckets.map(Number),
+                }).sendMeasurement(Number(body.measurement));
+                break;
+
+            default:
+                assertUnreachable(body.vdaf);
+        }
+        console.log("Successful upload");
+        res.send({"status": "success"});
+    } catch (error) {
+        console.error(error);
+        res.send({"status": "error", "error": String(error)});
```

Comment:
Should this have a non-2xx http status? As is, I believe this would result in a 200 status with "status": "error"

Reply:
Hmm, actually I should separate out different error conditions. All the input handling errors should be 400, but anything relating to errors uploading to the leader should be 200, with the error message part of the body as here.

This is covered in this paragraph: https://divergentdave.github.io/draft-dcook-ppm-dap-interop-test-design/draft-dcook-ppm-dap-interop-test-design.html#section-4-2. The intent is to keep test API-related errors separate from errors arising from the system under test. This is for the sake of test cases which expect DAP-level errors.